keymanagementscenarios.md

```diff
@@ -0,0 +1,139 @@
+
+Deployment Use Cases:
+
+Provisioning Truststore for Deployment
+
+![Notaryv2 Deployment Provisioning](./media/nv2-keymgmt-architecture-provisiontruststore-v1.png)
+
+Deployment Workflow
+
+![Notaryv2 Deployment Workflow](./media/nv2-keymgmt-architecture-pull-v1.png)
+
+Detailed Usecases:
+- Specify trusted public root keys
+    - Deployer Admin gets root public key from publisher.
+    - Deployer Admin adds root public key to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Specify location of trusted public root keys
+    - Deployer Admin gets location of key repository from publisher.
+    - Deployer Admin adds location of key repository to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Enterprises can restrict users to add/remove their root public keys to the trust store used across a fleet of runtime hosts. The trust store will validate images pulled from any registry.
+- Enterprises can restrict users to add/remove root public keys for trusted third parties to their trust stores. The trust store will validate images pulled from any registry.
+- Trust store also includes subordinate keys
+- Air gapped environments
+
+Key rotation/revocation use cases:
+
+Key Revocation Workflow
+
+![Notaryv2 Revocation Workflow](./media/nv2-keymgmt-architecture-revokekeys-v1.png)
+
+- Root Revocation (compromised root should not be needed in process to designate itself as revoked, otherwise attacker can use compromised root for a key rotation locking out publisher)
+    - Publisher deletes revoked root public key on registry (registry can stop sharing containers with revoked key).
+    - [TODO: Discuss if we want this] Registry stops vending containers signed with old root key (will prevent revoked artifacts from being used by developers not checking signature).
+    - Publisher removes revoked root public key from shared list (other users stop trusting artifacts with revoked key, provides defense in depth as they are not relying on registry).
+    - Publisher creates new root key following any scenario listed earlier in Signing use cases (starting from scratch as old key can no longer be trusted).
+    - Publisher uploads new root public key to registry (registry can verify user for container uploads). 
+    - Publisher updates shared root public key (other users can verify containers with new key before running).
+    - Publisher creates new delegate keys following any scenario listed earlier in Signing use cases.
+- Delegate Key Revocation (root can be used here for designating delegate keys as revoked as it has not been compromised)
+    - Publisher lists revoked delegate keys as untrusted. Revocation list is signed by root key (Design discussion on whether we use CRLs or some other approach to sharing information on revoked keys).
```

Comment:
How do we prevent a replay attack of the revocation list? (Replaying the revocation list from before a key was removed)